smarts/core/bubble_manager.py

```diff
@@ -563,9 +563,13 @@ def _prepare_sensors_for_agent_control(
         # Setup mission (also used for observations)
         # XXX: this is not quite right.  route may not be what the agent wants to take.
         route = sim.traffic_sim.vehicle_route(vehicle_id=vehicle.id)
+        if len(route) > 1:
+            goal = PositionalGoal.from_road(route[-1], sim.scenario.road_map)
+        else:
+            goal = EndlessGoal()
```

Comment:
What's the relevance of this with TraCI connection issue?

Reply:
The route is returned from the traffic simulation and can now default to `[]`. Thanks for pointing it out because I realise now that the length check for the first case should be `len(route)>0`.